src/main/services/csfloat/csfloatSocket.ts

```diff
@@ -0,0 +1,576 @@
+    } catch (error) {
+      console.error(`failed to cancel unconfirmed trade offers`, error);
+    }
+
+    try {
+      await this.pingTradeHistory(tradesInPending, tradeOffers);
+    } catch (error) {
+      console.error("failed to ping trade history", error);
+      errors.history_error = (error as any).toString();
+    }
+
+    try {
+      await this.pingSentTradeOffers(tradesInPending, tradeOffers);
+    } catch (error) {
+      console.error("failed to ping sent trade offer state", error);
+      errors.trade_offer_error = (error as any).toString();
+    }
+
+    try {
+      await this.pingCancelTrades(tradesInPending, tradeOffers);
+    } catch (e) {
+      console.error("failed to ping cancel ping trade offers", e);
+    }
+
+    return errors;
+  }
+
+  private async reportBlockedBuyers(
+    tradesInPending: ITradeFloat[],
+    ignoredOrBlokedUsers: string[]
+  ): Promise<void> {
+    if (!tradesInPending || tradesInPending.length < 1) return;
+
+    const hasTrade = tradesInPending.some(
+      (trade) =>
+        trade.seller_id.toString() === this.steamIDBase64 ||
+        trade.buyer_id === this.steamIDBase64
+    );
+
+    if (!hasTrade) return;
+
+    const filteredIDs = ignoredOrBlokedUsers.filter((steamID) => {
+      return tradesInPending.some(
+        (trade) => trade.seller_id == steamID || trade.buyer_id == steamID
+      );
+    });
+
+    if (filteredIDs.length === 0) return;
+
+    await this._csFloatClient.pingBlockedUsers(ignoredOrBlokedUsers);
+  }
+
+  private async cancelUnconfirmedTradeOffers(
+    tradesInPending: ITradeFloat[],
+    tradeOffers: IGetTradeOffersResponde
+  ): Promise<void> {
+    const offerIDsToCancel = [
+      ...new Set(
+        tradesInPending
+          .filter(
+            (trade) =>
+              trade.steam_offer.state ===
+              ETradeOfferStateCSFloat.CreatedNeedsConfirmation &&
+              new Date(trade.steam_offer.sent_at).getTime() <
+              Date.now() - 60 * 60 * 1000
+          )
+          .map((trade) => trade.steam_offer.id)
+      ),
+    ];
+
+    if (offerIDsToCancel.length === 0) return;
+
+    const tradeOffersConcatenated = tradeOffers.sent.concat(
+      tradeOffers.received
+    );
+
+    const offersIDsStillNeedsConfirmation = offerIDsToCancel.filter((id) => {
+      const sentOffer = tradeOffersConcatenated.find(
+        (offer) => offer.id === id
+      );
+      if (!sentOffer) return false;
+
+      return (
+        sentOffer.state === ETradeOfferStateCSFloat.CreatedNeedsConfirmation
+      );
+    });
+
+    if (offersIDsStillNeedsConfirmation.length === 0) return;
+
+    for (const offerID of offersIDsStillNeedsConfirmation) {
+      try {
+        await this.emit("cancelTrade", offerID);
+      } catch (error: any) {
+        console.error(
+          `failed to cancel needs confirmation trade, returning early: ${error.toString()}`
+        );
+        return;
+      }
```

Comment:
Event emition does not throw error.
```suggestion
      this.emit("cancelTrade", offerID);
```